### PR TITLE
bug fixes and tests

### PR DESCRIPTION
**Pull request by marci73**

### CODE REVIEW

File: src/cp_control_utils.F [hunk 1]
```diff
@@ -368,9 +368,9 @@ SUBROUTINE read_dft_control(dft_control, dft_section)
       CALL section_vals_get(tmp_section, n_repetition=nrep, explicit=is_present)
       IF (is_present) THEN
          ALLOCATE (dft_control%efield_fields(nrep))
-         DO i = 1, nrep
+!         DO i = 1, nrep
             CALL read_efield_sections(dft_control, tmp_section)
-         END DO
+!         END DO
```
Comment (dev-zero) [inline src/cp_control_utils.F]: why not remove it completely?

File: src/motion/md_run.F [hunk 2]
```diff
@@ -459,7 +459,13 @@ SUBROUTINE qs_mol_dyn_low(md_env, md_section, motion_section, force_env, globenv
          END IF
          IF (ehrenfest_md) force_env%qs_env%rtp%istep = istep
 
-         CALL cp_iterate(logger%iter_info, last=(istep == simpar%nsteps), iter_nr=itimes)
+!         CALL cp_iterate(logger%iter_info, last=(istep == simpar%nsteps), iter_nr=itimes)
```
Comment (dev-zero) [inline src/motion/md_run.F]: dito

File: src/qs_scf.F [hunk 3]
```diff
@@ -541,6 +541,11 @@ SUBROUTINE scf_env_do_scf(scf_env, scf_control, qs_env, converged, should_stop)
 
       IF (.NOT. converged) CPWARN("SCF run NOT converged")
 
+      IF (.NOT. converged .AND. scf_control%stop_higher_iter_level) THEN
+        logger%iter_info%last_iter(logger%iter_info%n_rlevel-1) = .TRUE.
+        CPWARN("MD iteration also stops")
```
Comment (dev-zero) [inline src/qs_scf.F]: this functionality does not seem to be `MD` exclusive and `scf_env_do_scf` is also called by `xas_tp_scf`

File: src/cp_control_utils.F (hunk 1 above)
Comment (marci73) [reply src/cp_control_utils.F]: yes right

File: src/motion/md_run.F (hunk 2 above)
Comment (marci73) [reply src/motion/md_run.F]: also in this case

File: src/qs_scf.F (hunk 3 above)
Comment (marci73) [reply src/qs_scf.F]: I will check, but should not be effective there

**Comment by dev-zero** [comment]
@oschuett Git(Hub) is luckily smart enough to detect the replicated commit and automatically removed it upon rebase+merge.

**Comment by oschuett** [comment]
Cool! Didn't know that :-)

**Pull request merged**